src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java

```diff
@@ -321,12 +327,14 @@ public ListBoxModel doFillRepositoryItems(@AncestorInPath SCMSourceOwner context
             }
             try {
                 GitHub github = Connector.connect(apiUri, Connector.lookupScanCredentials(context, apiUri, scanCredentialsId));
+
                 GHMyself myself = null;
                 try {
                     myself = github.getMyself();
                 } catch (IllegalStateException e) {
-                    // may be anonymous... ok to ignore
+                    LOGGER.log(Level.WARNING, e.getMessage());
```

Comment:
Well, in the UI we are showing "Credentials are required". Additionally, in GH Enterprise, if `private mode` is enabled annonymous access is not allowed.